ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -600,6 +600,11 @@ private JoinNode buildJoin(final Join root, final String prefix, final boolean i
 
     final JoinKey joinKey = buildJoinKey(root);
 
+    Optional<ColumnName> leftJoinNonKeyColumn = Optional.empty();
+    if (isForeignKeyJoin) {
+      leftJoinNonKeyColumn = getLeftJoinNonKeyColumn(root);
```

Comment:
Could we embed this into `verifyJoin` and change it's return type from `boolean` to `Optional<ColumnName>` ? If the optional is empty, it's a PK join, otherwise it's a FK-join? (Or would this be too messy?)

Reply:
If my understanding in my comment above regarding the use of qualifiers in `getLeftJoinNonKeyColumn()` is correct and that code is indeed not needed, then `getLeftJoinNonKeyColumn()` becomes very simple and my vote is also to embed it into `verifyJoin()` as Matthias has suggested.

Reply:
Done. It was easier because now I return `return Optional.of(((ColumnReferenceExp) leftExpression).getColumnName());` in the `verifyJoin`. I'd like to rename that method now, but I'm not sure what name to use. It verifies a join is correct, either PK or FK, but also returns the FK column name? what name to use? Maybe we can do the rename later.